### ISSUE TITLE
Repository projects: search filters per column

### ISSUE THREAD

**Issue by xarg** [isaacs/github#993]
For our organization the repo Projects are not very useful. The reason they are not is because we have to manually drag and drop the issues into each column. We're already using labels, statuses, etc.. to tag our issues. It would be super useful to be able to have a search filter for each column so that they are automatically populated.

Imagine a Kanban style project with columns: 

- Backlog `state: open`
- Ready `state: open label: ready`
- In Progress: `state: open label:in-progress`
- Done: `state: closed`

These filters could also be used to get a view on the overall distribution of the issue types. For example:

- All bugs `state:open label:bug`
- High priority bugs `state:open label:bug label:high`
- Features `state:open label:features`
- Integrations `state:open label:integrations`
- Ideas `state:open label:idea`

Or if you have a huge component based project like kubernetes you could see issues filtered by components to get a better understanding about which ones are very popular.

Note: we already use waffle.io, and it's useful, but you can only have a single board per repo.

Related: #566 

There remains the question of what happens when you drag & drop the issues from one column to the other. We would prefer something similar to what waffle.io has meaning that it removes the labels of the issue from the old column and adds new labels when it gets to the new column.

**Comment by danfoley** [comment]
Has anyone written a JavaScript based solution that could do this? Site like userscripts.org using code monkey etc.. It seems that a javascript tool could be written to do just this.

If not, I would love to start one. Would anyone be interested in joining me?

**Comment by xarg** [comment]
@danfoley I think part of it could be done using the [projects api](https://developer.github.com/v3/projects/) where you could define your columns with the filters in the name of the Column. Then have webhook server that listens on `issue-created` or `issue-updated` which  basically just manages the cards for you based on the filter that the can find in the name like this:

![screen shot 2017-09-21 at 9 23 10 am](https://user-images.githubusercontent.com/94721/30706890-a084cff2-9eae-11e7-9420-c1f101a43f32.png)

This would require to have an app registered on github and get permissions to the projects and the issues. I was thinking of doing this for a while, but didn't have the time.

Btw: https://github.com/marketplace/category/project-management

**Comment by danfoley** [comment]
Agreed. Time is the issue of course. If anyone starts it up, please link to the repo here so we can all jump in and help out.

**Comment by CWTAsk-Blog** [comment]
Okay for the filter ignoring when you start a project (i'm completely new to these things), what does every one of them mean it can do?

**Comment by mogul** [comment]
Just happened to run across this in the course of researching something else... You might be interested in this:
https://github.com/affrae/probot-column-issue-label-sync
